.github/workflows/ci.yml

```diff
@@ -42,7 +42,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v1
         with:
-          python-version: 3.12.1
+          python-version: 3.11.7
```

Comment:
is downgrading python intentional?

Reply:
Yeah 😬 

I had bumped it a bit ambitiously to 3.12, but upon running into some intial errors with a method of authenticating with a google cloud service in 3.12 (iirc, it's now been just long enough for the exact details to have escaped my memory) I opted to bump the python version down one little bit to 3.11, which resolved things.